### PR TITLE
WIP - Error message on invalid config file

### PR DESCRIPTION
**Pull request by pykong**
**TODO**:
- [x] Make all tests green. 

Note: Merge #55 first.

Closes: #57

### CODE REVIEW

File: tests/test_config.py [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+    expected = Path("tests/user_data.ref.txt").read_text()
+    assert result == expected
+
+
+def test_bad_toml(capsys):
+    with pytest.raises(InvalidConfigFileError):
+        load_toml_data("tests/demo_badtoml")
+
+
+def test_invalid_toml(capsys):
```
Comment (pykong) [inline tests/test_config.py]: What is the purposes of `test_invalid_toml` versus `test_bad_toml`?
What is the difference between an _invalid_ and a _bad_ config file in this context?
Why does `test_invalid_toml` call not the loaders for `yaml` and `json` as well?

Comment (jpscaletti) [reply tests/test_config.py]: 🤔

Comment (jpscaletti) [reply tests/test_config.py]: `test_bad_toml` test what happens when `config.toml` is not a valid TOML file.
But It seems it was a mistake adding it, because the badly named `test_invalid_toml` test all different config formats.

TL;DR:
- `test_bad_toml` and the `test/demo_badtoml` foldr should be **removed**.
- `test_invalid_toml`should be **replaced** by the following code

```python
def test_invalid_yaml(capsys):
    assert {} == load_yaml_data("tests/demo_invalid")
    out, _ = capsys.readouterr()
    assert re.search(r"INVALID.*tests/demo_invalid/copier\.yml", out)


def test_invalid_toml(capsys):
    assert {} == load_toml_data("tests/demo_invalid")
    out, _ = capsys.readouterr()
    assert re.search(r"INVALID.*tests/demo_invalid/copier\.toml", out)


def test_invalid_json(capsys):
    assert {} == load_json_data("tests/demo_invalid")
    out, _ = capsys.readouterr()
    assert re.search(r"INVALID.*tests/demo_invalid/copier\.json", out)


def test_invalid_data(capsys):
    assert {} == load_config_data("tests/demo_invalid", _warning=False)
    out, _ = capsys.readouterr()
    assert re.search(r"INVALID", out)
```

Comment (pykong) [reply tests/test_config.py]: @jpscaletti Thanks for your input. I am going to make this PR merge ready this weekend.

**Comment by pykong** [comment]
@jpscaletti Thanks for your input!
Tests are green. Please check my changes out!
